embedding/openai_test.go

```diff
@@ -0,0 +1,52 @@
+package embedding
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestOpenaiEmbeddings(t *testing.T) {
+	t.Parallel()
+
+	if openaiKey := os.Getenv("OPENAI_API_KEY"); openaiKey == "" {
```

Comment:
(separate from this) we should probably refactor these checks to be a clean little testutil helper